pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -84,9 +84,11 @@ func (pa *PodAutoscaler) annotationFloat64(key string) (float64, bool) {
 // not set.
 // Note: min will be ignored if the PA is not reachable
 func (pa *PodAutoscaler) ScaleBounds(asConfig *autoscalerconfig.Config) (int32, int32) {
-	var min int32
+	min := asConfig.MinScale
```

Comment:
I think this needs to be in the if block one line down, otherwise we'll keep minScale > 0 even when the revision becomes unreachable, which isn't how specifying min-scale-as-an-annotation acts